main.cpp

```diff
@@ -178,6 +195,27 @@ struct DoubleType
         DoubleType& pow( const DoubleType& operand );
         DoubleType& pow( const IntType& operand );
         DoubleType& pow( double operand );
+    
+        DoubleType& apply(std::function<DoubleType&(double&)> callable)
+        {
+            if(callable)
+            {
+                return callable(*heapDub); 
+            }
+            else
+            {
+                return *this;
+            }
```

Comment:
You have no code that runs after the `else` block.  Therefore you do not need this `else` keyword
- [ ] revise